wordpress-mobile/WordPress-iOS
# Fix Quick Start 'Follow other sites' tour

Fixes #13068 

In iOS13, `ManyDelegateNavigationController` no longer worked, causing `tourGuide.visited(.readerBack)` to not be called. Meaning the tour was never notified the readerBack step was completed.

To fix this:
- `ManyDelegateNavigationController` has been removed.
- `QuickStartNavigationWatcher` is now `QuickStartNavigationSettings`, and is no longer a `UINavigationControllerDelegate`.
- `QuickStartNavigationSettings` is updated in `WPSplitViewController`'s `navigationController:willShow`.

To test:
- Follow the repro steps noted on #13068 . (Thank you @rachelmcr for excellent repro steps.)
  - Hack: you can show the tour checklist by returning `true` in `QuickStartTourGuide:shouldShowChecklist`.
- Verify:
  - After tapping the Reader tab, the nav bar button `< Reader` is spotlighted.
    - In split view (like on an iPad) `< Reader` is not shown, so this step is skipped (this is existing functionality).
  - The `Tap < Reader to continue` hint is dismissed.
  - When you finish the tour (i.e. by tapping Reader > Search) the tour is marked as complete.


PR submission checklist:
